src/natcap/invest/spec_utils.py

```diff
@@ -217,3 +218,308 @@ def fallback_serializer(obj):
+        lines that are indented, that describe details of the arg such as
+        vector fields and geometries, option_string options, etc.
+    """
+    type_string = format_type_string(spec['type'])
+    in_parentheses = [type_string]
+
+    # For numbers and rasters that have units, display the units
+    units = None
+    if spec['type'] == 'number':
+        units = spec['units']
+    elif spec['type'] == 'raster' and spec['bands'][1]['type'] == 'number':
+        units = spec['bands'][1]['units']
+    if units:
+        units_string = format_unit(units)
+        if units_string and units_string != 'none':
+            in_parentheses.append(f'units: **{units_string}**')
+
+    if spec['type'] == 'vector':
+        in_parentheses.append(format_geometries_string(spec["geometries"]))
+
+    # Represent the required state as a string, defaulting to required
+    # It doesn't make sense to include this for boolean checkboxes
+    if spec['type'] != 'boolean':
+        # get() returns None if the key doesn't exist in the dictionary
+        required_string = format_required_string(spec.get('required'))
+        in_parentheses.append(f'*{required_string}*')
+
+    # Nested args may not have an about section
+    if 'about' in spec:
+        about_string = f': {spec["about"]}'
+    else:
+        about_string = ''
+
+    first_line = f"**{name}** ({', '.join(in_parentheses)}){about_string}"
+
+    # Add details for the types that have them
+    indented_block = []
+    if spec['type'] == 'option_string':
+        # may be either a dict or set. if it's empty, the options are
+        # dynamically generated. don't try to document them.
+        if spec['options']:
+            if isinstance(spec['options'], dict):
+                indented_block.append(_('Options:'))
+                indented_block += format_options_string_from_dict(spec['options'])
+            else:
+                formatted_options = format_options_string_from_list(spec['options'])
+                indented_block.append(_('Options:') + f' {formatted_options}')
+
+    elif spec['type'] == 'csv':
+        if 'columns' not in spec and 'rows' not in spec:
+            first_line += _(
+                ' Please see the sample data table for details on the format.')
+
+    # prepend the indent to each line in the indented block
+    return [first_line] + ['\t' + line for line in indented_block]
+
+
+def document_arg_rst(module_name, *arg_keys):
+    """Get the spec dictionary for a particular arg.
+
+    Args:
+        module_name (str): invest model module containing the arg.
+        *arg_keys: one or more strings that are nested arg keys.
+
+    Returns:
+        arg spec (dict)
+    """
+    # import the specified module (that should have an ARGS_SPEC attribute)
+    try:
+        module = importlib.import_module(module_name)
+        spec = module.ARGS_SPEC['args']
+    except ImportError:
+        raise ValueError(
```

Comment:
I thought it was a bit clearer when I was updating the user's guide and I got an error from misspelling a module. Since an `ImportError` is usually a problem in the code, but a `ValueError` suggests an incorrect value was provided. But I don't feel strongly about it.

Reply:
Yeah I see where you're coming from now. Still I think it's safer with the original exceptions, just in case an `ImportError` arises from some other cause at some point. Of course many other things would also come crashing down if that happened.